File: Entrada de Dados/Numerica/numero.scala
import scala.io.StdIn
import scala.util.control.Breaks._ 

object Main {
  def main(args: Array[String]): Unit = {
	  
    var NUMERO: Int = 0
    var NUMERO_AUX: String = ""

    breakable {
      do {
        print("Entre com um numero: ")
        NUMERO_AUX = StdIn.readLine()
        try {
          NUMERO = NUMERO_AUX.toInt
          if (NUMERO != 0) {break()}
        } catch {
          case _: NumberFormatException =>
            println("Entrada invalida. Por favor, tente novamente.")
        }
      } while (true)
    }
    println(s"Voce informou o número: $NUMERO")
  }
}

File: Entrada de Dados/Numerica/numero.java
import java.util.Scanner;

class Main {
  public static void main(String[] args) {

    int NUMERO;
    String NUMERO_AUX;

    Scanner entrada = new Scanner(System.in);
    do {
      System.out.print("Entre um numero: ");
      NUMERO_AUX = entrada.nextLine();
      try {
        NUMERO = Integer.parseInt(NUMERO_AUX);
        break;
      } catch (NumberFormatException e) {
        System.out.println("Entrada invalida. Por favor, tente novamente.");
      }
    } while (true);
    entrada.close();
    System.out.println("Voce informou o numero: " + NUMERO);
  }
}
